### PR TITLE
Add new accessibility sanitizer

### LINKED ISSUE #4528: Add sanitizer to ensure 'role' and 'tabindex' attributes are present for elements with 'on' attribute
## Feature description

Something that comes up over and over again in the support forums is people adding `on` handler attributes for `tap` events but then not also adding `role=button` and `tabindex=0` (actually, the value doesn't matter, just `role` and `tabindex`). Examples:

* https://wordpress.org/support/topic/the-attribute-tabindex-in-tag-span-is-missing-or-incorrect/
* https://wordpress.org/support/topic/amp-next-page-v1-strange-behaviour/#post-12625539
* https://wordpress.org/support/topic/search-console-error-with-amp-plugin/#post-12445708
* https://wordpress.org/support/topic/tabindex-bug/

In #4447 I proposed to add validator support to _remove_ the elements that lack these attributes. However, this would be somewhat jarring, especially in Reader mode where we don't yet have validation error reporting (#4480).

Therefore, I suggest a better behavior would be to introduce a new `AMP_Accessibility_Sanitizer` that looks for `//*[ @on and contains( @on, 'tap:' ) and ( not( @tabindex ) or not( @role ) )]` and then automatically supply the `role=button` if no `role` attribute is present, or `tabindex=0` if no `tabindex` is present. Note that this should only be done for elements that are not intrinsically focusable, for example `button` and `a` would not be included. The specific elements that don't require `role` and `tabindex` attributes are indicated in the spec [like for the `button` element](https://github.com/ampproject/amphtml/blob/2af70090cd66527d88d2aaeb1cae37864fb6d8ac/validator/validator-main.protoascii#L4031-L4038):

```
  attrs: {
    name: "role"
    implicit: true
  }
  attrs: {
    name: "tabindex"
    implicit: true
  }
```

See the [`on` attribute spec](https://github.com/ampproject/amphtml/blob/2af70090cd66527d88d2aaeb1cae37864fb6d8ac/validator/validator-main.protoascii#L6137-L6144):

```
  attrs: {
    name: "on"
    trigger: {
      if_value_regex: "tap:.*"
      also_requires_attr: "role"
      also_requires_attr: "tabindex"
    }
  }
```

Ideally these would be extracted from the spec to add to `AMP_Allowed_Tags_Generated` so that we don't have to hard-code them. This would be an ultimate prerequisite for adding support for the `trigger` construct in #4447, but it can/should be done up first for a minor release.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

### PR DESCRIPTION
**Pull request by schlessera**
## Summary

This PR includes the following changes:

- Adds a new `AMP_Accessibility_Sanitizer` class that adds `role` and `tabindex` attributes to elements with the `on` AMP event.
- Ports over the `MarkupComparison` trait from the `amp/common` test suite and replaces copy-pasted `assertEqualMarkup()` methods throughout tests with the new trait.
- Adds a `Role` interface to define and document ARIA roles and use the provided constants instead of hard-coded strings in existing sanitizers/transformers that define ARIA roles.

Fixes #4528

## Checklist

- [x] My pull request is addressing an [open issue](https://github.com/ampproject/amp-wp/contributing/project-management.md#life-of-an-issue) (please create one otherwise).
- [x] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/contributing/engineering.md#tests).
- [x] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/contributing/engineering.md) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

**Review by pierlon — CHANGES_REQUESTED** [review]
You forgot the most important part, adding it to the list of sanitizers to be run:

```diff
--- includes/amp-helper-functions.php	(revision f0c97bbfaddbc59f44e0c9508aa63d1be367a912)
+++ includes/amp-helper-functions.php	(date 1586320714141)
@@ -1220,6 +1220,7 @@
 		'AMP_Style_Sanitizer'             => [],
 		'AMP_Meta_Sanitizer'              => [],
 		'AMP_Layout_Sanitizer'            => [],
+		'AMP_Accessibility_Sanitizer'     => [],
 		'AMP_Tag_And_Attribute_Sanitizer' => [], // Note: This whitelist sanitizer must come at the end to clean up any remaining issues the other sanitizers didn't catch.
 	];
 ```

**Review by westonruter — APPROVED** [review]

**Review by pierlon — APPROVED** [review]

File: tests/php/test-class-amp-accessibility-sanitizer.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ * Tests the accessibility sanitizer class.
+ */
+class AMP_Accessibility_Sanitizer_Test extends WP_UnitTestCase {
+
+	use MarkupComparison;
+
+	public function get_sanitize_test_data() {
+		return [
+			'valid markup remains unchanged'            => [
+				'<div href="#" on="tap:some_id.toggleClass(some_class)" role="button" tabindex="0"></div>',
```
Comment (amedina) [inline tests/php/test-class-amp-accessibility-sanitizer.php:20]: @schlessera The validator complains against `href="#"`; it is not valid markup.

Comment (amedina) [reply tests/php/test-class-amp-accessibility-sanitizer.php:20]: Does the test catch that?

**Comment by amedina** [comment]
Would like to validate the acceptance criteria here. These HTML blocks:

Screen Shot 2020-04-14 at 10 55 05 AM

Should yield the corresponding `role` and/or `tabindex` attributes

Screen Shot 2020-04-14 at 10 57 47 AM

**Comment by kienstra** [comment]
This looks good in my local.

On copying the Custom HTML markup above verbatim into Custom HTML blocks:

```html
<div on="tap:myfunction()">
None in
</div>

<div on="tap:myfunction()" tabindex="1">
Tabindex in
</div>

<div on="tap:myfunction()" role="button">
Role in
</div>
```

...the `tabindex` and `role` appear as expected:

tabindex

**Comment by amedina** [comment]
Thanks @kienstra ; reason was missing `composer install` run

**Pull request merged**